### PR TITLE
networkctl: Add support to display ipvlan

### PR DESCRIPTION
**Pull request by ssahani**
```
❯ build/networkctl status myipvlan1                                                                                                                                                        ─╯
● 26: myipvlan1                                                 
             Link File: /usr/lib/systemd/network/99-default.link
          Network File: n/a                                     
                  Type: ether                                   
                 State: off (unmanaged)                         
                Driver: ipvlan                                  
            HW Address: 4e:c5:88:28:c1:c0                       
                   MTU: 1500 (min: 68, max: 65535)              
                 QDisc: noop                                    
                  Mode: L2 (bridge)                             
  Queue Length (Tx/Rx): 1/1    

```

### CODE REVIEW

**Comment by ssahani** [comment]
cc @yuwata

File: src/network/netdev/ipvlan.c [hunk 1]
```diff
@@ -4,25 +4,11 @@
 
 #include "conf-parser.h"
 #include "ipvlan.h"
+#include "ipvlan-util.h"
 #include "networkd-link.h"
 #include "string-table.h"
```
Comment (yuwata) [inline src/network/netdev/ipvlan.c]: Please drop 'string-table.h'.

File: src/network/networkctl.c [hunk 2]
```diff
@@ -1644,6 +1652,32 @@ static int link_status_one(
                                    TABLE_UINT16, info->tunnel_port);
                 if (r < 0)
                         return table_log_add_error(r);
+
+        } else if (streq_ptr(info->netdev_kind, "ipvlan")) {
+                _cleanup_free_ char *p = NULL, *s = NULL;
+
+
```
Comment (yuwata) [inline src/network/networkctl.c]: Doubled empty lines.

File: src/network/networkctl.c [hunk 3]
```diff
@@ -1644,6 +1652,32 @@ static int link_status_one(
...
+                if (info->ipvlan_flags & IPVLAN_F_PRIVATE)
+                        p = strdup("private");
+                else if (info->ipvlan_flags & IPVLAN_F_VEPA)
+                        p = strdup("vepa");
+                else
+                        p = strdup("bridge");
+
```
Comment (yuwata) [inline src/network/networkctl.c]: Please drop this empty line.

File: src/network/networkctl.c [hunk 4]
```diff
@@ -1644,6 +1652,32 @@ static int link_status_one(
...
+                if (!s)
+                        return log_oom();
+
+                r = table_add_many(table,
+                                   TABLE_EMPTY,
+                                   TABLE_STRING, "Mode:",
+                                   TABLE_STRING, s);
+                if (r < 0)
+                        return table_log_add_error(r);
+
```
Comment (yuwata) [inline src/network/networkctl.c]: Unnecessary empty line.

File: src/shared/ipvlan-util.c [hunk 5]
```diff
@@ -0,0 +1,23 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <net/if.h>
+
+#include "conf-parser.h"
```
Comment (yuwata) [inline src/shared/ipvlan-util.c]: This is not necessary.

Comment (ssahani) [reply src/shared/ipvlan-util.c]: `DEFINE_STRING_TABLE_LOOKUP` ?

**Comment by ssahani** [comment]
updated thanks for the review @yuwata . PTAL

File: src/shared/ipvlan-util.c (hunk 5 above)
Comment (poettering) [reply src/shared/ipvlan-util.c]: @yuwata is right, this include is not necessary. string-table.h makes DEFINE_STRING_TABLE_LOOKUP available, and that's all we need

File: src/shared/ipvlan-util.h [hunk 6]
```diff
@@ -0,0 +1,27 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#pragma once
+
+#include <netinet/in.h>
+#include <linux/if_link.h>
+
```
Comment (poettering) [inline src/shared/ipvlan-util.h:6]: this should include macro.h, so that `_pure_` and `_const_` used below are definitely defined.

**Comment by poettering** [comment]
also ,needs rebase

**Comment by ssahani** [comment]
Updated thanks for the review . PTAL.

File: src/network/networkctl.c [hunk 7]
```diff
@@ -1741,6 +1750,26 @@ static int link_status_one(
                                    TABLE_EMPTY,
                                    TABLE_STRING, "Mode:",
                                    TABLE_STRING, macvlan_mode_to_string(info->macvlan_mode));
```
Comment (poettering) [inline src/network/networkctl.c:1752]: the error checking of this table_add_many() call somehow got lost

**Pull request merged**